staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -159,7 +159,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	StorageVersionHash:      {Default: true, PreRelease: featuregate.Beta},
 	WatchBookmark:           {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 	APIPriorityAndFairness:  {Default: false, PreRelease: featuregate.Alpha},
-	RemoveSelfLink:          {Default: false, PreRelease: featuregate.Alpha},
+	RemoveSelfLink:          {Default: true, PreRelease: featuregate.Beta},
```

Comment:
Wait, this was already here? Do we not test feature gates in both positions in tests?

Did it work before? If it didn't work before, then no one could possibly have tested it and it's not clear to me that it's fair or safe to go to beta.

Reply:
Apparently not in unit tests...

The feature gate was there and the self-linking logic itself was gated.

Reply:
Actually - it seems that only a few e2e tests are running with all Alpha features enabled:
https://github.com/kubernetes/test-infra/blob/03cbb16ef550658d61fb14912d3dd19b1cd7fa21/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L428

This is why all of these weren't caught...